packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -1771,4 +2002,288 @@ describe('src/cy/commands/cookies', () => {
+          expect(lastLog.get('error').message).to.contain('some err message')
+          expect(lastLog.get('error')).to.eq(err)
+
+          done()
+        })
+
+        cy.clearAllCookies()
+      })
+    })
+
+    describe('.log', () => {
+      beforeEach(function () {
+        cy.on('log:added', (attrs, log) => {
+          if (attrs.name === 'clearAllCookies') {
+            this.lastLog = log
+          }
+        })
+
+        Cypress.automation
+        .withArgs('get:cookies', {})
+        .resolves([{ name: 'foo', domain: 'localhost' }])
+        .withArgs('clear:cookies', [{ name: 'foo', domain: 'localhost' }])
+        .resolves([
+          { name: 'foo' },
+        ])
+      })
+
+      it('can turn off logging', () => {
+        cy.clearAllCookies({ log: false }).then(function () {
+          expect(this.log).to.be.undefined
+        })
+      })
+
+      it('ends immediately', () => {
+        cy.clearAllCookies().then(function () {
+          const { lastLog } = this
+
+          expect(lastLog.get('ended')).to.be.true
+          expect(lastLog.get('state')).to.eq('passed')
+        })
+      })
+
+      it('snapshots immediately', () => {
+        cy.clearAllCookies().then(function () {
+          const { lastLog } = this
+
+          expect(lastLog.get('snapshots').length).to.eq(1)
+          expect(lastLog.get('snapshots')[0]).to.be.an('object')
+        })
+      })
+
+      it('#consoleProps', () => {
+        cy.clearAllCookies().then(function (cookies) {
+          expect(cookies).to.be.null
+          const c = this.lastLog.invoke('consoleProps')
+
+          expect(c['Yielded']).to.eq('null')
+          expect(c['Cleared Cookies']).to.deep.eq([{ name: 'foo' }])
+          expect(c['Num Cookies']).to.eq(1)
+        })
+      })
+    })
+
+    describe('.log with no cookies returned', () => {
+      beforeEach(function () {
+        cy.on('log:added', (attrs, log) => {
+          if (attrs.name === 'clearAllCookies') {
+            this.lastLog = log
+          }
+        })
+
+        Cypress.automation
+        .withArgs('get:cookies')
+        .resolves([])
+      })
+
+      it('#consoleProps', () => {
+        cy.clearAllCookies().then(function (cookies) {
+          expect(cookies).to.be.null
+          const c = this.lastLog.invoke('consoleProps')
+
+          expect(c['Yielded']).to.eq('null')
+          expect(c['Cleared Cookies']).to.be.undefined
+          expect(c['Note']).to.eq('No cookies were found or removed.')
+        })
+      })
+    })
+
+    describe('.log when no cookies were cleared', () => {
+      beforeEach(function () {
+        cy.on('log:added', (attrs, log) => {
+          if (attrs.name === 'clearAllCookies') {
+            this.lastLog = log
+          }
+        })
+
+        Cypress.automation
+        .withArgs('get:cookies', {})
+        .resolves([{ name: 'foo', domain: 'localhost' }])
+        .withArgs('clear:cookies', [{ name: 'foo', domain: 'localhost' }])
```

Comment:
What are the cases where clearAllCookies wouldn't clear a specific cookie?

Reply:
There aren't really. I updated this tests to better reflect reality. Addressed in https://github.com/cypress-io/cypress/pull/25012/commits/6c04ba02ab6155c83f8158808d3998d8a0b29f36